test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/function/BiPredicateTest.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.util.function
+
+import java.util.function.BiPredicate
+
+import org.junit.Assert._
+import org.junit.Test
+
+import org.scalajs.testsuite.utils.AssertThrows._
+
+class BiPredicateTest {
+  import BiPredicateTest._
+
+  private val largerThan10LessThan30 = makeBiPredicate[Int, Int](_ > 10, _ < 30)
+  private val even = makeBiPredicate[Int, Int](_ % 2 == 0, _ % 2 == 0)
+
+  private val throwingPredicate =
+    makeBiPredicate[Any, Any](x => throw new ThrowingPredicateException(x), _ => true)
+
+  private val dontCallPredicate =
+    makeBiPredicate[Any, Any](x => throw new AssertionError(s"dontCallPredicate.test($x)"), _ => true)
+
+  @Test def and(): Unit = {
+    // Truth table
+    val evenAndInRange = largerThan10LessThan30.and(even)
+
+    assertTrue("Both In Range and Even", evenAndInRange.test(22, 22))
+    assertTrue("Both In Range and Even", evenAndInRange.test(40, 22))
+
+    // Both In Range and Not Even
+    assertFalse("Both In Range and Not Even: Left Odd", evenAndInRange.test(41, 22))
+    assertFalse("Both In Range and Not Even: Right Odd", evenAndInRange.test(42, 21))
+    assertFalse("Both In Range and Not Even: Both Odd", evenAndInRange.test(41, 21))
```

Comment:
This still tests the predicates you create for the test in addition of the code under test.